codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/customizations/RulesBasedAuthSchemeResolverGeneratorTests.kt

```diff
@@ -0,0 +1,151 @@
+                            sigV4Option.signingProperties.set(key: AttributeKeys.signingName, value: "weather")
+                            guard let region = serviceParams.region else {
+                                throw ClientError.authError("Missing region in auth scheme parameters for SigV4 auth scheme.")
+                            }
+                            sigV4Option.signingProperties.set(key: AttributeKeys.signingRegion, value: region)
+                            validAuthOptions.append(sigV4Option)
+                        case "onlySigv4AuthOptional":
+                            var sigV4Option = AuthOption(schemeID: "aws.auth#sigv4")
+                            sigV4Option.signingProperties.set(key: AttributeKeys.signingName, value: "weather")
+                            guard let region = serviceParams.region else {
+                                throw ClientError.authError("Missing region in auth scheme parameters for SigV4 auth scheme.")
+                            }
+                            sigV4Option.signingProperties.set(key: AttributeKeys.signingRegion, value: region)
+                            validAuthOptions.append(sigV4Option)
+                            validAuthOptions.append(AuthOption(schemeID: "smithy.api#noAuth"))
+                        case "onlyCustomAuth":
+                            validAuthOptions.append(AuthOption(schemeID: "com.test#customAuth"))
+                        case "onlyCustomAuthOptional":
+                            validAuthOptions.append(AuthOption(schemeID: "com.test#customAuth"))
+                            validAuthOptions.append(AuthOption(schemeID: "smithy.api#noAuth"))
+                        default:
+                            var sigV4Option = AuthOption(schemeID: "aws.auth#sigv4")
+                            sigV4Option.signingProperties.set(key: AttributeKeys.signingName, value: "weather")
+                            guard let region = serviceParams.region else {
+                                throw ClientError.authError("Missing region in auth scheme parameters for SigV4 auth scheme.")
+                            }
+                            sigV4Option.signingProperties.set(key: AttributeKeys.signingRegion, value: region)
+                            validAuthOptions.append(sigV4Option)
+                    }
+                    return validAuthOptions
+                }
+
+                public func constructParameters(context: HttpContext) throws -> ClientRuntime.AuthSchemeResolverParameters {
+                    return try DefaultS3AuthSchemeResolver().constructParameters(context: context)
+                }
+            }
+
+            public struct DefaultS3AuthSchemeResolver: S3AuthSchemeResolver {
+                public func resolveAuthScheme(params: ClientRuntime.AuthSchemeResolverParameters) throws -> [AuthOption] {
+                    var validAuthOptions = Array<AuthOption>()
```

Comment:
nit should use `[AuthOption]()`

Reply:
Changed as suggested, not just for rules based, but for model based auth resolver generation as well.